pkg/common/common_test.go

```diff
@@ -67,3 +71,159 @@ func TestFind(t *testing.T) {
 		t.Error("should not have found anything in the slice")
 	}
 }
+
+func TestFetchEnv(t *testing.T) {
+	t.Run("when env var exists & has a value different from the default value then return the env var value", func(t *testing.T) {
+		key := "LOG_LEVEL"
+		defaultVal := "info"
+		val := "debug"
+		os.Setenv(key, val)
+		defer os.Unsetenv(key)
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != val {
+			t.Errorf("Expected %v, but got %v", val, result)
+		}
+	})
+	t.Run("when env var exists but has an empty value then return the empty value", func(t *testing.T) {
+		key := "LOG_MODE"
+		defaultVal := "production"
+		val := ""
+		os.Setenv(key, val)
+		defer os.Unsetenv(key)
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != val {
+			t.Errorf("Expected %v, but got %v", val, result)
+		}
+	})
+	t.Run("when env var does not exist & the default value is used then return the default value", func(t *testing.T) {
+		key := "LOG_MODE"
+		defaultVal := "production"
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != defaultVal {
+			t.Errorf("Expected %v, but got %v", defaultVal, result)
+		}
+	})
+	t.Run("when a runtime error occurs during execution then return the default value", func(t *testing.T) {
+		key := "LOG_LEVEL"
+		defaultVal := "info"
+
+		os.Unsetenv(key)
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != defaultVal {
+			t.Errorf("Expected %v, but got %v", defaultVal, result)
+		}
+	})
+	t.Run("when default value is an empty string then return an empty string", func(t *testing.T) {
+		key := "LOG_MODE"
+		defaultVal := ""
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != defaultVal {
+			t.Errorf("Expected %v, but got %v", defaultVal, result)
+		}
+	})
+}
+
+func TestGetDefaultIfNil(t *testing.T) {
+	t.Run("when value is non-nil pointer type and default value is provided then return value", func(t *testing.T) {
+		val := "test"
+		def := "default"
+
+		result := GetDefaultIfNil(&val, def)
+
+		if result != val {
+			t.Errorf("Expected %v, but got %v", val, result)
+		}
+	})
+	t.Run("when value is nil pointer type and default value is provided then return default value", func(t *testing.T) {
+		var val *string
+		def := "default"
+
+		result := GetDefaultIfNil(val, def)
+
+		if result != def {
+			t.Errorf("Expected %v, but got %v", def, result)
+		}
+	})
+	t.Run("when value is not a pointer type and default value is provided then return default value", func(t *testing.T) {
```

Comment:
this test is somewhat a duplication. The value is non nil pointer, hence the value of `val` (0 in the test) must be returned.

Reply:
I see, it is alrady initialised with a default value (in this case 0), so technically it is an initialised non-nil value. I'll delete this one as well then!